src/main/java/util/ApiQuotaManager.java

```diff
@@ -0,0 +1,181 @@
+    }
+
+    /**
+     * Restarts the refreshTimer.
+     */
+    public void restartRefreshTimer(){
+        this.refreshTimer.restart();
+    }
+
+    /**
+     * Updates the period of the refreshTimer for synchronization of the data store.
+     * The initial call to this method does not update the period of the refreshTimer.
+     * @param e RefreshTimerTriggeredEvent object that holds the current API quota information.
+     */
+    private void updateSyncRefreshRate(RefreshTimerTriggeredEvent e) {
+        ApiQuotaInfo info = e.getApiQuotaInfo();
+
+        if (remainingApiRequests == UNINITIALIZED){
+            remainingApiRequests = info.getRemainingRequests();
+            return;
+        }
+
+        apiCallsUsedInPreviousRefresh = computeApiCallsUsedInPreviousRefresh(info.getRemainingRequests());
+        remainingApiRequests = info.getRemainingRequests();
+        refreshDurationInMinutes = computeRefreshTimerPeriod(info.getRemainingRequests(),
+                                                             Utility.minutesFromNow(info.getNextRefreshInMillisecs()),
+                                                             apiCallsUsedInPreviousRefresh, API_QUOTA_BUFFER,
+                                                             DEFAULT_REFRESH_PERIOD_IN_MINS);
+        this.refreshTimer.restartTimer((int) Utility.minsToSecs(refreshDurationInMinutes));
+        logger.info("Refresh period updated to " + refreshDurationInMinutes
+                + "mins with API calls used in previous refresh cycle is " + apiCallsUsedInPreviousRefresh
+                + ", current API quota is " + info.getRemainingRequests() + " and next API quota top-up in "
+                + info.getNextRefreshInMinutesFromNow() + "mins.");
+    }
+
+    /**
+     * Computes the API calls used in previous refresh.
+     * @param newRemainingApiRequests The number of API requests remaining in the current API quota window.
+     * @return The number of API calls used in previous refresh or apiCallsUsedInPreviousRefresh.
+     *         apiCallsUsedInPreviousRefresh will be returned when newRemainingApiRequests > remainingApiRequests.
+     */
+    private int computeApiCallsUsedInPreviousRefresh(int newRemainingApiRequests) {
+        int apiUsageInRecentRefresh = remainingApiRequests - newRemainingApiRequests;
+
+        if (apiUsageInRecentRefresh < 0) {
+            return apiCallsUsedInPreviousRefresh;
+        }
+
+        return apiUsageInRecentRefresh;
+    }
+
+    /**
+     * Computes the TickerTimer period that is used for refreshing the issues periodically
+     * Assumes future refreshes will take the same number of API calls as the last refresh and finds out the refresh
+     * duration that will spread out the refreshes until the next API quota top up.
+     *
+     * Calculation avoids scheduling a refresh that happens during the same time as the API quota top up.
+     *
+     * PMD warning is suppressed to allow explicit parenthesis.
```

Comment:
PMD warning suppression is an implementation detail, so there is no need to include it in the header comment (i.e. that detail is not relevant to users of the method  -- header comment is for the users of the method)